server/pulp/server/webservices/views/content.py

```diff
@@ -476,3 +481,134 @@ def delete(self, request, upload_id):
         upload_manager = factory.content_upload_manager()
         upload_manager.delete_upload(upload_id)
         return generate_json_response(None)
+
+
+class ContentSourceView(View):
+    """
+    View for content sources.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        Get all content sources.
+
+        :param request:   WSGI request object, body contains bits to upload
+        :type  request:   django.core.handlers.wsgi.WSGIRequest
+
+        :return: list of sources
+        :rtype:  django.http.HttpResponse
+        """
+        container = ContentContainer()
+        sources = []
+        for source in container.sources.values():
+            d = source.dict()
+            link = {'_href': reverse('content_sources_resource',
+                    kwargs={'source_id': source.id})}
+            d.update(link)
+            sources.append(d)
+        return generate_json_response_with_pulp_encoder(sources)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_allow_empty
+    def post(self, request, action):
```

Comment:
Sharing views for more than one url has the same problem as [this one](https://pulp.plan.io/issues/824)

I think it would be pretty simple to just split this into 2 views.